### DISCUSSION TITLE
I want to install URLHub in a DockerContainer with php8.3

### DISCUSSION THREAD

**Discussion by csaeum** [realodix/urlhub#1030] [Q&A] [answered]
I want to install URLHub in a DockerContainer with php8.3

I have a PHP8.3 container with Apache and NODE22
I want to install URLHub on it.

I have several projects running in this type of container that I have put together (e.g. Shopware 5 or 6 or Contao and Matomo)

So generally my technology works.

But I'm having problems with URLHub right now, so I hope you can help me.

I'm basically following the instructions on GitHub and executing the commands in the PHP container itself as the user "docker", who is assigned all the directories and rights.

Here is my approach:

In the .env I only filled the MySQL and aka DB_ variables as well as my APP_URL aka domain
The APP_ENV=prod

`The command: composer install`
has no errors:

> > @php artisan filament:upgrade
>  ⇂ public/js/filament/widgets/components/chart.js
>  ⇂ public/js/filament/widgets/components/stats-overview/stat/chart.js
>  ⇂ public/js/filament/support/async-alpine.js
>  ⇂ public/js/filament/support/support.js
>  ⇂ public/css/filament/support/support.css
> 
>  INFO Successfully published assets!
>  INFO Configuration cache cleared successfully.
>  INFO Route cache cleared successfully.
>  INFO Compiled views cleared successfully.
>  INFO Successfully upgraded!
> 
> 122 packages you are using are looking for funding.
> Use the `composer fund` command to find out more!
> > @php artisan vendor:publish --tag=laravel-assets --ansi --force
> 
>  INFO No publishable resources for tag [laravel-assets].

`The command: php artisan key:generate`

> runs without problems

`The command: php artisan migrate --seed`
`runs without problems`

Since I don't want to run it locally, I next run:
`The command: npm install`

but this ends with these messages:

> docker@f420602c216c:/var/www/html/urlhub-Projekt$ npm install
> 
> npm error code ERESOLVE
> 
> npm error ERESOLVE unable to resolve dependency tree
> 
> npm error
> 
> npm error While resolving: undefined@undefined
> 
> npm error Found: vite@5.4.11
> 
> npm error node_modules/vite
> 
> npm error dev vite@"^5.2.10" from the root project
> 
> npm error
> 
> npm error Could not resolve dependency:
> 
> npm error peer vite@"^6.0.0" from laravel-vite-plugin@1.1.0
> npm error node_modules/laravel-vite-plugin
> npm error dev laravel-vite-plugin@"^1.0.2" from the root project
> npm error
> npm error Fix the upstream dependency conflict, or retry
> npm error this command with --force or --legacy-peer-deps
> npm error to accept an incorrect (and potentially broken) dependency resolution.
> npm error
> npm error
> npm error For a full report see:
> npm error /home/docker/.npm/_logs/2024-12-02T20_23_39_659Z-eresolve-report.txt
> npm error A complete log of this run can be found in: /home/docker/.npm/_logs/2024-12-02T20_23_39_659Z-debug-0.log
> docker@f420602c216c:/var/www/html/urlhub-project$

I want to install URLHub in a DockerContainer with php8.3

I have a PHP8.3 container with Apache and NODE22
I want to install URLHub on it.

I have several projects running in this type of container that I have put together (e.g. Shopware 5 or 6 or Contao and Matomo)

So generally my technology works.

But I'm having problems with URLHub right now, so I hope you can help me.

I'm basically following the instructions on GitHub and executing the commands in the PHP container itself as the user "docker", who is assigned all the directories and rights.

Here is my approach:

In the .env I only filled the MySQL and aka DB_ variables as well as my APP_URL aka domain
The APP_ENV=prod

The command: composer install
has no errors:

> @php artisan filament:upgrade
 ⇂ public/js/filament/widgets/components/chart.js
 ⇂ public/js/filament/widgets/components/stats-overview/stat/chart.js
 ⇂ public/js/filament/support/async-alpine.js
 ⇂ public/js/filament/support/support.js
 ⇂ public/css/filament/support/support.css

 INFO Successfully published assets!
 INFO Configuration cache cleared successfully.
 INFO Route cache cleared successfully.
 INFO Compiled views cleared successfully.
 INFO Successfully upgraded!

122 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> @php artisan vendor:publish --tag=laravel-assets --ansi --force

 INFO No publishable resources for tag [laravel-assets].

The command: php artisan key:generate
runs without problems

The command: php artisan migrate --seed

runs without problems

Since I don't want to run it locally, I next run:
The command: npm install

but this ends with these messages:

> 
> docker@f420602c216c:/var/www/html/urlhub-Projekt$ npm install
> 
> npm error code ERESOLVE
> 
> npm error ERESOLVE unable to resolve dependency tree
> 
> npm error
> 
> npm error While resolving: undefined@undefined
> 
> npm error Found: vite@5.4.11
> 
> npm error node_modules/vite
> 
> npm error dev vite@"^5.2.10" from the root project
> 
> npm error
> 
> npm error Could not resolve dependency:
> 
> npm error peer vite@"^6.0.0" from laravel-vite-plugin@1.1.0
> npm error node_modules/laravel-vite-plugin
> npm error dev laravel-vite-plugin@"^1.0.2" from the root project
> npm error
> npm error Fix the upstream dependency conflict, or retry
> npm error this command with --force or --legacy-peer-deps
> npm error to accept an incorrect (and potentially broken) dependency resolution.
> npm error
> npm error
> npm error For a full report see:
> npm error /home/docker/.npm/_logs/2024-12-02T20_23_39_659Z-eresolve-report.txt
> npm error A complete log of this run can be found in: /home/docker/.npm/_logs/2024-12-02T20_23_39_659Z-debug-0.log
> docker@f420602c216c:/var/www/html/urlhub-project$

Where is my mistake?

**Comment by realodix** [comment, marked as answer]
> npm error peer vite@"^6.0.0" from laravel-vite-plugin@1.1.0
> npm error node_modules/laravel-vite-plugin
> npm error dev laravel-vite-plugin@"^1.0.2" from the root project

I think this is similar to https://github.com/laravel/vite-plugin/pull/310#issuecomment-2512565467

**Comment by csaeum** [comment]
Hello and thanks for your Response

I think I'm much further along now, but unfortunately I keep getting the http:// URL in the source code.

It's true that URLHub doesn't run "locally" with http, but my Traefik uses HTTPS for "global".

What and where do I have to enter the "trusted_proxies" as they are called in Symfony, for example?

**Comment by realodix** [comment]
Configuring Trusted Proxies? You can see https://laravel.com/docs/11.x/requests#configuring-trusted-proxies
